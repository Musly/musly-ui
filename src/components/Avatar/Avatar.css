.avatar {
  position: relative;
  overflow: hidden;

  @apply rounded-md;
}

.avatar > .badge {
  position: absolute;
  width: 12px;
  height: 12px;
  right: -0.125rem;
  bottom: -0.125rem;

  @apply border-2 border-white rounded-full;
}

.avatar > .badge--online {
  @apply bg-green-500;
}

.avatar > .badge--offline {
  @apply bg-gray-400;
}
