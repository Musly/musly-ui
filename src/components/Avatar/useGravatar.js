import querystring from 'query-string';
import isRetina from 'is-retina';
import md5 from 'md5';

const baseUrl = '//www.gravatar.com/avatar/';

export function useGravatar (email, size = 256) {
  const formattedEmail = `${email}`.trim().toLowerCase();
  const hash = md5(formattedEmail);
  const query = querystring.stringify({ s: size, r: 'g', d: 'mp' });
  const retinaQuery = querystring.stringify({ s: size * 2, r: 'g', d: 'mp' });

  return isRetina() ? `${baseUrl}${hash}/?${retinaQuery}` : `${baseUrl}${hash}/?${query}`;
}
