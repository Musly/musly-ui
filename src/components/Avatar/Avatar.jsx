import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './Avatar.css';
import { useGravatar } from './useGravatar';

export function Avatar ({ email, title, online }) {
  const badgeClasses = classNames({
    badge: true,
    'badge--online': online,
    'badge--offline': !online,
  });

  const url = useGravatar(email, 88);

  return (
    <div className="avatar">
      <img
        src={url}
        alt={title || email}
        className="h-auto"
        width="44"
        height="44"
      />
      <div className={badgeClasses} data-test-id="badge" />
    </div>
  );
}

Avatar.propTypes = {
  email: PropTypes.string.isRequired,
  title: PropTypes.string,
  online: PropTypes.bool,
};

Avatar.defaultProps = {
  title: null,
  online: false,
};
