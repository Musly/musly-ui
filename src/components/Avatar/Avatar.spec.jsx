import React from 'react';
import { shallow } from 'enzyme';
import { Avatar } from './Avatar';

describe('<Avatar />', () => {
  it('should render the Avatar with offline badge', () => {
    const result = shallow(<Avatar email="test@test.de" />);
    expect(result.find('img').props().alt).toEqual('test@test.de');
    expect(result.find('[data-test-id="badge"]').hasClass('badge--offline')).toBe(true);
    expect(result.find('[data-test-id="badge"]').hasClass('badge--online')).toBe(false);
    expect(result).toMatchSnapshot();
  });

  it('should render the Avatar with online badge', () => {
    const result = shallow(<Avatar email="test@test.de" title="John Doe" online />);
    expect(result.find('img').props().alt).toEqual('John Doe');
    expect(result.find('[data-test-id="badge"]').hasClass('badge--offline')).toBe(false);
    expect(result.find('[data-test-id="badge"]').hasClass('badge--online')).toBe(true);
    expect(result).toMatchSnapshot();
  });
});
