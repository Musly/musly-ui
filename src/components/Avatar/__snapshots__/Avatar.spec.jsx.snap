// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render the Avatar with offline badge 1`] = `
<div
  className="avatar"
>
  <img
    alt="test@test.de"
    className="h-auto"
    height="44"
    src="//www.gravatar.com/avatar/f84d37ce99493155ee296c2b746191d0/?d=mp&r=g&s=88"
    width="44"
  />
  <div
    className="badge badge--offline"
    data-test-id="badge"
  />
</div>
`;

exports[`<Avatar /> should render the Avatar with online badge 1`] = `
<div
  className="avatar"
>
  <img
    alt="John Doe"
    className="h-auto"
    height="44"
    src="//www.gravatar.com/avatar/f84d37ce99493155ee296c2b746191d0/?d=mp&r=g&s=88"
    width="44"
  />
  <div
    className="badge badge--online"
    data-test-id="badge"
  />
</div>
`;
